mempool/reactor.go

```diff
@@ -132,10 +133,10 @@ func (memR *Reactor) OnStart() error {
 	return nil
 }
 
-// GetChannels implements Reactor.
-// It returns the list of channels for this reactor.
+// GetChannels implements Reactor by returning the list of channels for this
+// reactor.
 func (memR *Reactor) GetChannels() []*p2p.ChannelDescriptor {
-	maxMsgSize := calcMaxMsgSize(memR.config.MaxTxBytes)
+	maxMsgSize := memR.config.MaxBatchBytes + protoOverheadForTxsMessage
```

Comment:
I don't believe this calculation is correct. Each tx in the list will be length-prefixed with a varint, so the Protobuf overhead depends on the size of each tx (which determines the size of the varint) and the number of txs.

Reply:
oh, you're right! maybe we can just use MaxBatchBytes and remove protoOverheadForTxsMessage

Reply:
Yes, as long as you take into account the protobuf overhead of each tx.

Reply:
Has this been resolved? It seems like the length prefixes aren't taken into account at all anymore.

Reply:
added a note saying MaxBatchBytes includes "space needed by encoding (one varint per transaction)"

Reply:
But that's not what the code does. If I set `MaxBatchBytes` to 5, and submit a tx `12345`, then this tx will be accepted by the `txs()` function - however, when encoded, this message will take 6 bytes, thus the message will be rejected and our connection will be dropped.

Reply:
wonder why tests didn't catch this